modules/perception/camera/app/cipv_camera.cc

```diff
@@ -0,0 +1,967 @@
+  // float center_y = static_cast<float>(object->center(1));
+  float center_x =
+    static_cast<float>(object->camera_supplement.local_center[0]);
+  float center_y =
+    static_cast<float>(object->camera_supplement.local_center[1]);
+  // Direction should be changed based on alpha angle and more.
+  float direction_x = object->direction(0);
+  float direction_y = object->direction(1);
+  float x1 = size_x / 2;
+  float x2 = 0 - x1;
+  float y1 = size_y / 2;
+  float y2 = 0 - y1;
+  float len = static_cast<float>(
+              sqrt(direction_x * direction_x + direction_y * direction_y));
+  float cos_theta = direction_x / len;
+  float sin_theta = -direction_y / len;
+
+  // Check if object direction is less than +-45 degree
+  // angle = atan2(y, x)
+  if (fabs(atan2(object->direction(1), object->direction(0))) <=
+      FOURTY_FIVE_DEGREE) {
+    // get back of the vehicle
+    closted_object_edge->start_point[0] =
+        x2 * cos_theta + y1 * sin_theta + center_x;
+    closted_object_edge->start_point[1] =
+        y1 * cos_theta - x2 * sin_theta + center_y;
+
+    closted_object_edge->end_point[0] =
+        x2 * cos_theta + y2 * sin_theta + center_x;
+    closted_object_edge->end_point[1] =
+        y2 * cos_theta - x2 * sin_theta + center_y;
+
+    // If a vehicle faces side way, extract the side edge of a vehicle
+  } else if (atan2(object->direction[1], object->direction[0]) >
+             FOURTY_FIVE_DEGREE) {
+    // get left side of the vehicle
+    closted_object_edge->start_point[0] =
+        x2 * cos_theta + y1 * sin_theta + center_x;
+    closted_object_edge->start_point[1] =
+        y1 * cos_theta - x2 * sin_theta + center_y;
+
+    closted_object_edge->end_point[0] =
+        x2 * cos_theta + y1 * sin_theta + center_x;
+    closted_object_edge->end_point[1] =
+        y1 * cos_theta - x2 * sin_theta + center_y;
+  } else if (atan2(object->direction[1], object->direction[0]) <
+             -FOURTY_FIVE_DEGREE) {
+    // get right side of the vehicle
+
+    closted_object_edge->start_point[0] =
+        x1 * cos_theta + y2 * sin_theta + center_x;
+    closted_object_edge->start_point[1] =
+        y2 * cos_theta - x1 * sin_theta + center_y;
+
+    closted_object_edge->end_point[0] =
+        x2 * cos_theta + y2 * sin_theta + center_x;
+    closted_object_edge->end_point[1] =
+        y2 * cos_theta - x2 * sin_theta + center_y;
+  } else {
+    // don't get front of vehicle
+    closted_object_edge->start_point[0] =
+        x1 * cos_theta + y1 * sin_theta + center_x;
+    closted_object_edge->start_point[1] =
+        y1 * cos_theta - x1 * sin_theta + center_y;
+
+    closted_object_edge->end_point[0] =
+        x1 * cos_theta + y2 * sin_theta + center_x;
+    closted_object_edge->end_point[1] =
+        y2 * cos_theta - x1 * sin_theta + center_y;
+  }
+
+  return true;
+}
+// Get closest edge of an object in ground coordinate
+// *** TO DO *** This function should be changed to find min-y and max-y edges
```

Comment:
`TODO(techoe): xxx.`

Reply:
Thank you will PR changes. I will also change 4 other places with "TO DO"